### ISSUE TITLE
Aliases defined under `resolve.alias` are duplicated

### ISSUE THREAD

**Issue by DanielHZhang** [solidjs/vite-plugin-solid#20]
`vite-plugin-solid` duplicates aliases listed under `resolve.alias`. Any downstream vite plugins that reference the user config in the `configResolved` hook have to deal with de-duplicating the user aliases.

Output result of `config.resolve.alias` in the `configResolved` hook after running `vite-plugin-solid`:
```
aliases: [
  { find: /^[\/]?@vite\/env/, replacement: [Function: replacement] },
  { find: /^[\/]?@vite\/client/, replacement: [Function: replacement] },      
  {
    find: 'src',
    replacement: 'C:\\Users\\Daniel\\Documents\\Workspace\\test\\src'
  },
  {
    find: 'shared',
    replacement: 'C:\\Users\\Daniel\\Documents\\Workspace\\test\\shared'
  },
  {
    find: 'src',
    replacement: 'C:\\Users\\Daniel\\Documents\\Workspace\\test\\src'
  },
  {
    find: 'shared',
    replacement: 'C:\\Users\\Daniel\\Documents\\Workspace\\test\\shared'
  },
  { find: /^solid-refresh$/, replacement: '/@solid-refresh' }
]
```

`vite.config.ts`:
```
import path from 'path';
import {defineConfig} from 'vite';
import solidPlugin from 'vite-plugin-solid';
import svgLoaderPlugin from './src/svg-loader';

export default defineConfig({
  resolve: {
    alias: [
      {
        find: 'src',
        replacement: path.join(process.cwd(), 'src'),
      },
      {
        find: 'shared',
        replacement: path.join(process.cwd(), 'shared'),
      },
    ],
  },
  plugins: [solidPlugin(), svgLoaderPlugin()],
  server: {
    port: 8080,
    strictPort: true,
  },
});
```

The issue might be from creating a new array on this line with `mergeAndConcat`:
https://github.com/solidjs/vite-plugin-solid/blob/48ca0c088145e1e4c10846245a2d1a50c471ee58/src/index.ts#L267

**Comment by amoutonbrady** [comment]
Hey! Thank you for reporting the issue. I'll look into it ASAP :)

**Comment by amoutonbrady** [comment]
Sorry for the delay, I added an extra test on the [./playground/vite.config.js](./playground/vite.config.js) config file and manage to reproduce and seemingly fix the issue. It appears, vite automatically already merge stuff together properly so I just removed the `mergeAndConcat` call.

It's been released in 2.1.0

Let me know if that works for you.

**Comment by DanielHZhang** [comment]
Yup can confirm that they're no longer being duplicated, thanks for the fix!
